src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/SelectionPropertyMetadataMapper.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata;
+
+use Sulu\Component\Content\Metadata\PropertyMetadata as ContentPropertyMetadata;
+use Webmozart\Assert\Assert;
+
+class SelectionPropertyMetadataMapper implements PropertyMetadataMapperInterface
+{
+    public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata): PropertyMetadata
+    {
+        $mandatory = $propertyMetadata->isRequired();
+        $minMaxValue = static::getValidatedMinMaxValue($propertyMetadata);
+
+        return new ArrayMetadata(
+            $propertyMetadata->getName(),
+            $mandatory,
+            new SchemaMetadata([], [
+                new SchemaMetadata([], [], [], 'string'),
+                new SchemaMetadata([], [], [], 'number'),
+            ]),
+            $minMaxValue->min,
+            $minMaxValue->max,
+            true
+        );
+    }
+
+    /**
+     * Result object has following structure:
+     * <code>
+     * ?int $min - value of min parameter
+     * ?int $max - value of max parameter
+     * </code>.
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function getValidatedMinMaxValue(
```

Comment:
What's the reason for this function to be static?

Reply:
I was just thinking... Isn't that the kind of stuff that would make sense in a separate `PropertyMetadataMapper`, e.g. a `MinMaxPropertyMetadataMapper`? And every `MetadataMapper` that needs that, can inject that one? Code could then look something like this:

```php
class SelectionPropertyMetadataMapper
{
    private $minMaxPropertyMetadataMapper;

    public function __construct(MinMaxPropertyMetadataMapper $minMaxPropertyMetadataMapper)
    {
        $this->minMaxPropertyMetadataMapper = $minMaxPropertyMetadataMapper;
    }

    public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata): PropertyMetadata
    {
        return $this->minMaxPropertyMetadataMapper
            ->mapPropertyMetadata($propertyMetadata) // returns a ArrayMetadata object
            ->withSchemaMetadata( // new function on the ArrayMetadataObject, that returns a new instance
                 new SchemaMetadata(
                     [],
                     [
                         new SchemaMetadata([], [], [], 'string'),
                         new SchemaMetadata([], [], [], 'number'),
                     ]
                 )
             );
    }
}
```

Reply:
Just for documentation: I didn't implement it this way, because we probably need the `MinMaxValueResolver` in more places in the future ... One example could be `min_characters` and `max_characters` of the `text_line` field type